terraform/modules/gce/compute/nodejs/nodejs.tf

```diff
@@ -0,0 +1,227 @@
+  lifecycle { create_before_destroy = true }
+
+  allow {
+    protocol = "tcp"
+    ports    = ["80", "443"]
+  }
+
+  allow {
+    protocol    = "icmp"
+    //TODO add tags to subnets
+  }
+}
+
+resource "aws_elb" "blue" {
+  name                        = "${var.name}-blue"
+  connection_draining         = true
+  connection_draining_timeout = 400
+
+  subnets         = ["${split(",", var.public_subnet_ids)}"]
+  security_groups = ["${aws_security_group.elb.id}"]
+
+  lifecycle { create_before_destroy = true }
+
+  listener {
+    lb_port           = 80
+    lb_protocol       = "http"
+    instance_port     = 8888
+    instance_protocol = "http"
+  }
+
+  listener {
+    lb_port            = 443
+    lb_protocol        = "https"
+    instance_port      = 8888
+    instance_protocol  = "http"
+    ssl_certificate_id = "${aws_iam_server_certificate.nodejs.arn}"
+  }
+
+  health_check {
+    healthy_threshold   = 2
+    unhealthy_threshold = 3
+    timeout             = 10
+    interval            = 15
+    target              = "HTTP:8888/"
+  }
+}
+
+resource "aws_elb" "green" {
+  name                        = "${var.name}-green"
+  connection_draining         = true
+  connection_draining_timeout = 400
+
+  subnets         = ["${split(",", var.public_subnet_ids)}"]
+  security_groups = ["${aws_security_group.elb.id}"]
+
+  lifecycle { create_before_destroy = true }
+
+  listener {
+    lb_port           = 80
+    lb_protocol       = "http"
+    instance_port     = 8888
+    instance_protocol = "http"
+  }
+
+  listener {
+    lb_port            = 443
+    lb_protocol        = "https"
+    instance_port      = 8888
+    instance_protocol  = "http"
+    ssl_certificate_id = "${aws_iam_server_certificate.nodejs.arn}"
+  }
+
+  health_check {
+    healthy_threshold   = 2
+    unhealthy_threshold = 3
+    timeout             = 10
+    interval            = 15
+    target              = "HTTP:8888/"
+  }
+}
+
+resource "terraform_remote_state" "aws_global" {
+  backend = "atlas"
+
+  config {
+    name = "${var.atlas_username}/${var.atlas_aws_global}"
+  }
+
+  lifecycle { create_before_destroy = true }
+}
+
+resource "template_file" "blue_user_data" {
+  template = "${path.module}/nodejs.sh.tpl"
+
+  lifecycle { create_before_destroy = true }
+
+  vars {
+    atlas_username    = "${var.atlas_username}"
+    atlas_environment = "${var.atlas_environment}"
+    atlas_token       = "${var.atlas_token}"
+    node_name         = "${var.name}"
+    deploy            = "blue"
+    site_ssl_cert     = "${var.site_ssl_cert}"
+    vault_ssl_cert    = "${var.vault_ssl_cert}"
+    vault_token       = "${var.vault_token}"
+    vault_policy      = "${var.vault_policy}"
+    aws_region        = "${var.region}"
+    aws_access_id     = "${element(split(",", terraform_remote_state.aws_global.output.iam_vault_access_ids), index(split(",", terraform_remote_state.aws_global.output.iam_vault_users), format("vault-%s", var.atlas_environment)))}"
```

Comment:
Can remove AWS specific stuff here too as we'll be removing it from the template_file.